### PR TITLE
server: do not chmod bind mounts with user namespaces

### PR DESCRIPTION
**Pull request by giuseppe**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

do not change the ownership of each parent directory in the source directory when user namespaces are used.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: server/sandbox_run_linux.go [hunk 1]
```diff
@@ -869,10 +864,9 @@ func (s *Server) runPodSandbox(ctx context.Context, req *types.RunPodSandboxRequ
 			g.AddMount(proc)
 		}
 		rootPair := sandboxIDMappings.RootPair()
-		for _, path := range pathsToChown {
-			if err := os.Chown(path, rootPair.UID, rootPair.GID); err != nil {
-				return nil, fmt.Errorf("cannot chown %s to %d:%d: %w", path, rootPair.UID, rootPair.GID, err)
-			}
+		path := sbox.ResolvPath()
+		if err := os.Chown(path, rootPair.UID, rootPair.GID); err != nil {
```
Comment (haircommander) [inline server/sandbox_run_linux.go:868]: can you add a comment why we need a Chwon here but not for the others?

Comment (giuseppe) [reply server/sandbox_run_linux.go]: it seems to not be required.  I've dropped it and simplified the patch further

**Comment by haircommander** [comment]
you've got some unit test failures. Do we still want this?

**Comment by giuseppe** [comment]
> you've got some unit test failures. Do we still want this?

yes I think it is still good to have.

Just rebased, let's see if it makes any difference

**Comment by giuseppe** [comment]
@haircommander CI is green

**Comment by giuseppe** [comment]
rebased

**Comment by giuseppe** [comment]
tests seem green now

**Comment by haircommander** [comment]
/approve

LGTM, @cri-o/cri-o-maintainers PTAL

**Comment by giuseppe** [comment]
@mrunalp PTAL

**Comment by giuseppe** [comment]
@cri-o/cri-o-maintainers PTAL

**Pull request merged**